server/src/routes/donatedItemRoutes.ts

```diff
@@ -55,17 +56,64 @@ router.post('/', donatedItemValidator, async (req: Request, res: Response) => {
 // GET /donatedItem - Fetch all donated items
 router.get('/', async (req: Request, res: Response) => {
     try {
-        const items = await prisma.donatedItem.findMany({
+    const donatedItem = await prisma.donatedItem.findMany({
+        include: {
+            donor: true, // Include all donor details
+            program: true, // Include all program details
+            statuses: { 
+                orderBy: {
+                    dateModified: 'asc' // Ensure they are ordered chronologically
+                }
+            }
+        }
+    });
+    res.json(donatedItem);
+} catch (error) {
+    if (error instanceof Error) {
+        console.error('Error fetching donated item:', error.message);
+        res.status(error.message.includes('must be an integer') ? 400 : 404).json({ error: error.message });
+    } else {
+        console.error('Error fetching donated item:', 'Unknown error');
+        res.status(500).json({ error: 'Unknown error' });
+    }
+    
+}
+});
+
+
+// GET /donatedItem - Fetch donated item by ID
+router.get('/:id', async (req: Request, res: Response) => {
+        const donatedItemId = parseInt(req.params.id);
+        if (isNaN(donatedItemId)) {
```

Comment:
you have defined `validateDonatedItem` function to validate id, you can call it here